### ISSUE TITLE
False positive "The variable does not seem to be defined for all execution paths"

### ISSUE THREAD

**Issue by nokitakaze** [scrutinizer-ci/php-analyzer#558]
> Here, you can explain why the found issue was invalid, and what result
> you would have expected otherwise.

As I can see boolean constants does not parsed at all. I can not say if my issue is the same as issue #366 

All functions in scrutinizer-test.php are marked as having this issue. 

see https://scrutinizer-ci.com/b/NokitaKaze/ascetcms/inspections/36bc84e9-003b-4fd9-bbdc-a3647e31099f/issues/files/scrutinizer-test.php?status=new&orderField=path&order=asc&issueId=6d38dafb-7f9a-490e-a946-81545f8be51d

**Comment by schmittjoh** [comment]
Thanks for reporting. This issue looks a bit crafted though. Could you elaborate why this `ìf`` condition is used? 

We could fix this, but it would add complexity that does not seem beneficial for real-world code.

**Comment by nokitakaze** [comment]
> Could you elaborate why this if condition is used?

Because this file's only purpose is to reproduce this issue. https://bitbucket.org/NokitaKaze/ascetcms/src/d3485602131666451e1617adfe3a051a41b5cb89/scrutinizer-test.php?at=scrutinizer

The real life usage is here https://bitbucket.org/NokitaKaze/ascetcms/src/83c256c5f7852f0356a8a6cda8447a2c01fbb250/functions/sql.php?at=concept#cl-85

**Comment by schmittjoh** [comment]
I see, so this issue looks more like this:

```php

function foo($a) {
    if ($a == 0) {
       // do something
    } else {
        $b = 'foo';
    }

    if ($a == 1) {
        return $b; // false-positive here
    }
}
```

This looks like a duplicate of #529.

**Comment by nokitakaze** [comment]
Yes. As you can see I started from your version and then simplified my function more and more.

And sorry I didn't find issue #529 because I was searching for string "The variable does not seem to be defined "

**Comment by schmittjoh** [comment]
Closing this here as we have the other issue still.
